pkg/controllers/manifestwork.go

```diff
@@ -100,17 +100,52 @@ func ScaffoldManifestwork(hyd *hypdeployment.HypershiftDeployment) (*workv1.Mani
 	return w, nil
 }
 
-func setManifestWorkSelectivelyDeleteOption(mw *workv1.ManifestWork, hostingNamespace string) {
-	mw.Spec.DeleteOption = &workv1.DeleteOption{
-		PropagationPolicy: workv1.DeletePropagationPolicyTypeSelectivelyOrphan,
-		SelectivelyOrphan: &workv1.SelectivelyOrphan{
-			OrphaningRules: []workv1.OrphaningRule{
-				{
-					Resource: "namespaces",
-					Name:     hostingNamespace,
+func setManifestWorkSelectivelyDeleteOption(mw *workv1.ManifestWork, hyd *hypdeployment.HypershiftDeployment) {
+	hostingNamespace := helper.GetHostingNamespace(hyd)
+
+	if hyd.Spec.Override == hypdeployment.InfraOverrideDestroy {
```

Comment:
If the override is Orphan, I think we should set the deleteOption to `DeletePropagationPolicyTypeOrphan`? this will make all resources(hostedclster, nodepool, secrets,ns) the manifestwork created orphan.